api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -125,12 +127,10 @@ Tokens getCurrentTokens() {
       throw new RuntimeException(e);
     } catch (ExecutionException e) {
       Throwable cause = e.getCause();
-      if (cause instanceof RuntimeException) {
-        throw (RuntimeException) cause;
-      } else if (cause instanceof Error) {
+      if (cause instanceof Error) {
         throw (Error) cause;
```

Comment:
We could do the same for `RuntimeException`. That should hopefully simplify (undo) recent changes in test code, WDYT?

Reply:
But I would like the stack trace to be populated here, so to inform the user where the error was thrown. If we do `throw (RuntimeException) e`, the stack trace will be that of the thread that instantiated the exception – which is pretty much meaningless for the user since they don't called that code directly.

Here is the result of throwing a new exception; note how the first printed stack clearly shows which API V2 method was being called:

```
java.lang.RuntimeException: Failed to acquire a valid OAuth2 access token
	at org.projectnessie.client.auth.oauth2.OAuth2Client.getCurrentTokens(OAuth2Client.java:133)
	at org.projectnessie.client.auth.oauth2.OAuth2Client.authenticate(OAuth2Client.java:118)
	at org.projectnessie.client.auth.oauth2.OAuth2AuthenticationProvider$OAuth2Authentication.addAuthHeader(OAuth2AuthenticationProvider.java:64)
	at org.projectnessie.client.http.impl.BaseHttpRequest.lambda$prepareRequest$0(BaseHttpRequest.java:68)
	...
	at org.projectnessie.client.http.HttpRequest.get(HttpRequest.java:80)
	...
	at org.projectnessie.client.http.impl.BaseHttpRequest.prepareRequest(BaseHttpRequest.java:68)
	at org.projectnessie.client.http.impl.jdk11.JavaRequest.executeRequest(JavaRequest.java:95)
	at org.projectnessie.client.http.HttpRequest.get(HttpRequest.java:80)
	at org.projectnessie.client.rest.v2.HttpGetAllReferences.get(HttpGetAllReferences.java:51)
  ...

Caused by: org.projectnessie.client.auth.oauth2.OAuth2Exception: OAuth2 server replied with HTTP status code 401 and error code "invalid_credentials": Try Again
	at org.projectnessie.client.auth.oauth2.OAuth2Client.checkErrorResponse(OAuth2Client.java:384)
	at org.projectnessie.client.http.impl.jdk11.JavaRequest.lambda$executeRequest$1(JavaRequest.java:140)
	at java.base/java.util.Collections$SingletonList.forEach(Collections.java:4856)
	at org.projectnessie.client.http.impl.jdk11.JavaRequest.executeRequest(JavaRequest.java:140)
	at org.projectnessie.client.http.HttpRequest.post(HttpRequest.java:90)
	at org.projectnessie.client.http.HttpRequest.postForm(HttpRequest.java:105)
	at org.projectnessie.client.auth.oauth2.OAuth2Client.fetchNewTokens(OAuth2Client.java:272)
  ...
	at org.projectnessie.client.auth.oauth2.OAuth2TokenRefreshExecutor$TokenRefreshThread.run(OAuth2TokenRefreshExecutor.java:56)

```

If we rethrow, we would get only the second stack above.

Reply:
I got it now, sorry about the confusion. Current PR LGTM :+1: